articles/send-sms-in-nodejs-using-vonage-api/index.md

```diff
@@ -0,0 +1,229 @@
+4. Since we will use Nodejs, select it as the programming language.
+5. Click on SMS, then wait for your account to be set up.
+
+You have successfully created a Vonage account; your dashboard should look like this.
+![Vonage's Dashboard](/engineering-education/flutter-folder-organization/dashboard.png)
+
+### Setting up out Node application
+
+#### Init application
+Setup up the node application by running the below command:
+```bash
+npm init
+```
+This command will create an empty package.json for our application. `package.json` holds the metadata about the project's dependencies, scripts, and version.
+
+#### Install dependencies
+Next, we will install the dependencies for our application. We will use `express` as our backend framework for Node, `EJS` as our template engine, `body-parser` to parse HTTP request body, `nexmo API` for sending the message, and `socket.io` to request a delivery report for our messages. To install the dependencies and save them to our package.json, run the below command in your terminal.
+```bash
+npm install –save express, nexmo, eggs, body-parser, socket.io, fetch
+```
+#### Nodemon
+We will need `nodemon` to continuously watch our application as we develop.  I prefer installing nodemon globally. To install nodemon, run the following command.
+```bash
+npm install -g nodemon
+```
+We are all set. Now let us get to the code!
+### Implementation
+First we need to create out applications entry point file. I will name this as `app.js`. This file will contain out driver code. You can set app’s entry point using the init command or manually set up within the package.json file. In out app.js, we will import our dependencies as shown below:
+```js
+const express = require('express');
+const bodyParser = require('body-parser');
+const ejs = require('ejs');
+const Nexmo = require('nexmo');
+const socketio = require('socket.io');
+const fetch = require("node-fetch");
+```
+
+Next, we initialize our application with express and set up our templating engine, EJS.  The EJS allows us to use `.html` extensions for our views. We will add the below code to our  `app.js`:
+```js
+const app = express()
+//Setting up the body parser middleware and ejs template engine
+app.set('view-engine', 'html');
+app.engine('html', ejs.renderFile);
+
+//public folder setup
+app.use(express.static(__dirname + '/public'));
+
+//body parser middleware
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended:true}));
+
+```
+
+Specifying our index route as our app will only have a single page `index.html`.
+```js
+//index route
+app.get('/', (request, response) =>{
+    response.render('index.html');
+});
+```
+### Creating our view
+We will create a folder called `views` where we will store our view file. Under the folder, we create an `index.html` file where we will have the input fields for holding the phone number and message to be sent. Here is how the `index.html` will look like:
```

Comment:
We will create a folder called `views` where we will store our view `file`   -  view files